### PR TITLE
Train options

### PR DESCRIPTION
**Pull request by wojtrz13**

### CODE REVIEW

File: API/Tensorflow/native_libs/src/c_wrappers/operations.cpp [hunk 1]
```diff
@@ -335,21 +335,98 @@ void** eval_graph_with_placeholders(GraphSession *graph,
...
+            std::shared_ptr<State>& state, uint32_t start_index, uint32_t end_index, bool apply_side_effects = true) {
+        std::map<std::string, std::shared_ptr<Tensor>> substitutions;
+        double running_loss = 0;
+        for (uint32_t input_index = start_index; input_index < end_index; ++input_index) {
+            for (size_t ph = 0; ph < ph_count; ++ph) {
+                substitutions.emplace(std::string(ph_names[ph]),
+                                      LifetimeManager::instance().accessOwned(ph_values[input_index * ph_count + ph]));
+            }
+            std::shared_ptr<EvaluationResult> eval_result = graph->eval(substitutions, state, apply_side_effects);
+            running_loss += to_double(eval_result->outputs[0], 0);
```
Comment (radeusgd) [inline API/Tensorflow/native_libs/src/c_wrappers/operations.cpp:399]: This will throw an exception when loss is not convertible to double as in `to_double`.
I don't think making it impossible at all to train models with other losses is a good idea.
I understand it's rather unlikely to have integer losses, but it's certainly possible so maybe it would be good to allow for this. One way might be to catch the exception and just ignore loss history?

File: API/Tensorflow/src/Tensor.luna [hunk 2]
```diff
@@ -58,11 +58,6 @@ class Tensor a:
         dims = self.dims
         flat = self.toFlatList
 
-        # Internal.
```
Comment (radeusgd) [inline API/Tensorflow/src/Tensor.luna]: Why is this documentation removed?

Comment (wojtrz13) [reply API/Tensorflow/src/Tensor.luna]: Because comments inside functions break something in Luna and these weren't too useful anyway
